web/pandas/pdeps/0007-copy-on-write.md

```diff
@@ -0,0 +1,580 @@
+   new DataFrame or Series, always _behaves as if_ it were a copy in terms of user
+   API.
+2. We implement Copy-on-Write (as implementation detail). This way, we can actually use
+   views as much as possible under the hood, while ensuring the user API behaves as a
+   copy.
+3. As a consequence, if you want to modify an object (DataFrame or Series), the only way
+   to do this is to modify that object itself directly.
+
+This addresses multiple aspects: 1) a clear and consistent user API (a clear rule: _any_
+subset or returned series/dataframe **always** behaves as a copy of the original, and
+thus never modifies the original) and 2) improving performance by avoiding excessive
+copies (e.g. a chained method workflow would no longer return an actual data copy at each
+step).
+
+Because every single indexing step behaves as a copy, this also means that with this
+proposal, "chained assignment" (with multiple setitem steps) will _never_ work and
+the `SettingWithCopyWarning` can be removed.
+
+## Background
+
+pandas' current behavior on whether indexing returns a view or copy is confusing. Even
+for experienced users, it's hard to tell whether a view or copy will be returned (see
+below for a summary). We'd like to provide an API that is consistent and sensible about
+returning views vs. copies.
+
+We also care about performance. Returning views from indexing operations is faster and
+reduces memory usage. The same is true for several methods that don't modify the data
+such as setting/resetting the index, renaming columns, etc that can be used in a method
+chaining workflow and currently return a new copy at each step.
+
+Finally, there are API / usability issues around views. It can be challenging to know
+the user's intent in operations that modify a subset of a DataFrame (column and/or row
+selection), like:
+
+```python
+>>> df = pd.DataFrame({"A": [1, 2], "B": [3, 4], "C": [5, 6]})
+>>> df2 = df[["A", "B"]]
+>>> df2.loc[df2["A"] > 1, "A"] = 1
+```
+
+Did the user intend to modify `df` when they modified `df2` (setting aside issues with
+the current implementation)? In other words, if we had a perfectly consistent world
+where indexing the columns always returned views or always returned a copy, does the
+code above imply that the user wants to mutate `df`?
+
+There are two possible behaviours the user might intend:
+
+1. Case 1: I know my subset might be a view of the original and I want to modify the
+   original as well.
+2. Case 2: I just want to modify the subset without modifying the original.
+
+Today, pandas' inconsistency means _neither_ of these workflows is really possible. The
+first is difficult, because indexing operations often (though not always) return copies,
+and even when a view is returned you sometimes get a SettingWithCopyWarning when
+mutating. The second is somewhat possible, but requires many defensive copies (to avoid
+SettingWithCopyWarnings, or to ensure that you have a copy when a view _was_ returned).
```

Comment:
```suggestion
and even when a view is returned you sometimes get a `SettingWithCopyWarning` when
mutating. The second is somewhat possible, but requires many defensive copies (to avoid
`SettingWithCopyWarning`s, or to ensure that you have a copy when a view _was_ returned).
```